### ISSUE TITLE
Require metaparameter in define resource is giving false positive.

### ISSUE THREAD

**Issue by sathlan** [voxpupuli/puppet-lint-absolute_classname-check#9]
Hi,

Given this snippet:

``` puppet
  create_resources($res,
    { "${name}" => {
      ensure             => $ensure,
      require            => Exec['do-it'],
    }
  })
```

and running 

```
puppet-lint /tmp/man.pp
```

then we get

```
WARNING: class included by relative name on line 4
```

This is linked to this issue/commit https://github.com/voxpupuli/puppet-lint-absolute_classname-check/commit/c4355ef833becd92c5063109e64543c652fd2a27

The version used is 0.2.2, but in this case the problem still persists.

**Comment by sathlan** [comment]
For reference, a workaround is this:

``` puppet
  create_resources($res,
    { "${name}" => {
      ensure             => $ensure,
     # lint:ignore:relative_classname_inclusion
     require            => Exec['do-it'],
     # lint:endignore
    }
  })
```

**Comment by bastelfreak** [comment]
Hi, we released 0.2.2 a few minutes ago. The release should include a bugfix for this, cna you check which version of this plugin you use?

**Comment by sathlan** [comment]
Hi, I use 0.2.2.

**Comment by sathlan** [comment]
The fix work for in general, but not when you use `create_resources`

**Comment by mvisonneau** [comment]
Yep! `0.2.2` fix doesn't work for `create_resources` :

``` puppet
$packages = {}
$defaults = {
   require => Exec['apt_update'],
}
create_resources( package, $packages, $defaults )
```

``` bash
manifest.pp - WARNING: class included by relative name on line n
```

**Comment by domcleal** [comment]
I'll see if I can make the check more precise, thanks for the report.

**Comment by domcleal** [comment]
#10 attempts to fix this by ignoring any function name (like require, include) it recognises when it's followed by a `=>` symbol. This ought to stop it picking up `require` in both hash keys and resource parameters (#6).

**Comment by claytono** [comment]
I'm also seeing a false positive in this case:

```
    User <| title == 'swift' |> {
      ensure     => present,
      system     => true,
      groups     => 'swift',
      home       => '/home/swift',
      managehome => false,
      shell      => $swift_shell,
      uid        => $swift_user_id,
      gid        => $swift_group_id,
      before     => Package['swift'],
      require    => Group['swift'],
    }

```

It's complaining that the require line is using a relative classname.

**Comment by domcleal** [comment]
@dvorak I've checked and your code snippet passes now with my open PR.

**Comment by traylenator** [comment]
Yes same here:

``` puppet
 File{
    ensure  => file,
    mode    => '0644',
    require => Package['shibboleth'],
    notify  => Service['shibd'],
    owner   => root,
    group   => root,
  }
```

**Comment by sathlan** [comment]
Hi @domcleal, I confirm that your patch works for the `create_resources` case.  Thank a lot for the quick fix.

**Comment by rnelson0** [comment]
This fix is available n 0.2.4.
